### DISCUSSION TITLE
How can I keep function/class name, just encrypt function/class content

### DISCUSSION THREAD

**Discussion by shaobo-zhang-senscomm** [dashingsoft/pyarmor#1734] [Q&A]
e.g.

```
class pyarmortest():
    def run(self, a, b):
        print(int(a)+int(b))
```

expecting:
```
class pyarmortest():
    def run(self, a, b):
b'PY000000\x00\x03\n\x00o\r\r\n\x80\x00\x01\x00\x08\x00\x00\x00\x04\x00\x00\x00@\x00\x00\x0083\x00\x00\x12\x01\x04\x00\xc75n\x89\xd8\xf3\x94\x18\xe1\xfe\xed\n\x85\xe8(\xcd\x00\x00\x00\x00\x00\x00\x00\x00Xu\x04v\xbe\xce\xaf\x85\x17\x1b]
```

Thanks

**Comment by jondy** [comment]
You need do nothing, function/class name isn't changed in obfuscated scripts. Just run an obfuscated script to verify it.

**Reply by shaobo-zhang-senscomm** [reply]
But I am using trial version, execute `pyarmor gen fib.py`, the file is as below.
`# Pyarmor 8.4.6 (trial), 000000, non-profits, 2024-04-12T14:55:52.797137 from pyarmor_runtime_000000 import __pyarmor__ __pyarmor__(__name__, __file__, b'PY000000\x00\x03\n\x00o\r\r\n\x80\x00\.+) `
Need a pro or group version?

**Comment by jondy** [comment]
You just can't see function/class name, but they can be used as plain scripts, it's the basic feature of pyarmor: using the obfuscated scriptssame as plain scripts in most of cases.

**Reply by unmurphy** [reply]
@jondy 
But my requirement is see the class/function name, because i need to manually import the obfuscated package's class/function name. otherwise, I wouldn't know what class/function this obfuscated package have.
Besides, it's seems that `obf_func=0` and `obf_class=0` might solve my problem, but it doesn't work. thank you~
****
谢谢~，我现在已经能将包混淆，打包并安装成功，运行也没有问题；但我的需求是在引入这个被混淆包的时候能知道这个包里面有哪些class/function name，但具体实现被混淆。我看文档里面有提到一些参数可以设置，比如 `obf_func=0` and `obf_class=0`，但经过测试还是全部混淆了。能帮忙看一下么，谢谢~

Pyarmor version: 8.5.11

**Comment by jondy** [comment]
If the obfuscated scripts i, try to builtin function `dir`. If the obfuscated scripts is obfuscated by restrict mode, you can't get them.

**Reply by unmurphy** [reply]
all right! so is there any other solution to resolve my problem?  such as change to any other mode, etc.

**Comment by jondy** [comment]
The obfuscated scripts are always like `__pyarmor__(xxxx)`, there is no any option could change its form.

The function/class name can be seen only by importing this obfuscated script, and find module attributes by `dir(modname)`
